### PR TITLE
STEAM-541 Predicting in R with a MOJO or POJO Jar file

### PR DESCRIPTION
**Pull request by mstensmo**
Adds h2o.predict_json in R with which you can run predictions on a mojo or pojo jar model directly without having h2o running.

### CODE REVIEW

File: h2o-r/h2o-package/R/predict.R [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#'
+#' H2O Prediction from R
```
Comment (jangorecki) [inline h2o-r/h2o-package/R/predict.R]: This name seems to be a little too generic for tor use case

File: h2o-r/h2o-package/R/predict.R [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#' @examples
+#' \donttest{
+#' library(h2o)
+#' h2o.predict_json('~/GBM_model_python_1473313897851_6.zip', '{"C7":1}')
+#' h2o.predict_json('~/GBM_model_python_1473313897851_6.zip', '{"C7":1}', c(".", "lib"))
+#' }
+#' @usage
+#' h2o.predict_json(model, json, classpath)
+#' @name h2o.predict_json
+
```
Comment (jangorecki) [inline h2o-r/h2o-package/R/predict.R]: As commented above, better to keep documentation comments (`#' ...`) in single chunk, then there is no need to repeat `@name h2o.predict_json`.
Additionally you are putting multiple variables into h2o package namespace, they will act as a global variables for any function in h2o namespace. None of those seems to be time consuming, I would simply put them inside the function. Unless you need to use them in another function?

File: h2o-r/h2o-package/R/predict.R [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+#' @param model  String with file name of MOJO or POJO Jar
+#' @param json  JSON String with inputs to model
+#' @param classpath  (Optional) Extra items for the class path of where to look for Java classes, e.g., h2o-genmodel.jar
+#' @return Returns an object with the prediction result
+#' @importFrom jsonlite fromJSON
+#' @examples
+#' \donttest{
+#' library(h2o)
+#' h2o.predict_json('~/GBM_model_python_1473313897851_6.zip', '{"C7":1}')
```
Comment (jangorecki) [inline h2o-r/h2o-package/R/predict.R]: Examples should be reproducible so we can test package with `--run-dontrun --run-donttest`. Best to use some tiny dataset (maybe iris) and very basic model.

File: h2o-r/h2o-package/R/predict.R [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	   h2o_predict_json_javaargs <- paste(" ", h2o_predict_json_javaopts, " -cp ", h2o_predict_json_classpath, " water.util.H2OPredictor", sep="")
+        }
+	if (h2o_predict_json_iswindows) { jsonq <- paste('"', json, '"', sep="") } else { jsonq <- paste("'", json, "'", sep="") } 
+	args <- paste(h2o_predict_json_javaargs, model, jsonq, sep=" ")
+	res <- system2(h2o_predict_json_java, args, stdout=TRUE, stderr=TRUE)
+	if (substring(res, 1, 1) == '{' || substring(res, 1, 1) == '[') {
+	   	json <- fromJSON(paste0(res, collapse=""))
+		return(json)
+	}
+	print(res)
```
Comment (jangorecki) [inline h2o-r/h2o-package/R/predict.R]: you should return `res`, print will happen automatically if `h2o.predict_json(.)` call results are not assigned to new variable.

File: h2o-r/h2o-package/R/predict.R [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+#' h2o.predict_json('~/GBM_model_python_1473313897851_6.zip', '{"C7":1}')
+#' h2o.predict_json('~/GBM_model_python_1473313897851_6.zip', '{"C7":1}', c(".", "lib"))
+#' }
+#' @usage
+#' h2o.predict_json(model, json, classpath)
+#' @name h2o.predict_json
+
+h2o_predict_json_java <- "java"
+h2o_predict_json_javaopts <- "-Xmx4g"
+h2o_predict_json_javapath <- c( c(".", "h2o-genmodel.jar"),  sapply(.libPaths(), paste0, "/h2o/java/h2o.jar", USE.NAMES=FALSE) )
```
Comment (jangorecki) [inline h2o-r/h2o-package/R/predict.R]: `/h2o/java/h2o.jar` looks platform specific, we have `.h2o.downloadJar` which will detect path to h2o.jar. Better to not duplicate functionality which already exists.

**Comment by ledell** [comment]
Follow up to my "approval" above -- don't merge until tests pass.

**Comment by mstensmo** [comment]
@ledell thanks for reviewing. @jangorecki please review again, all your suggestions have been addressed.

**Comment by mstensmo** [comment]
h2o-ops please test!

**Comment by mstensmo** [comment]
h2o-ops please test!

**Comment by abal5** [comment]
h2o-ops please test!

**Comment by mstensmo** [comment]
h2o-ops please test!

**Comment by abal5** [comment]
@h2o-ops please test!

**Comment by ledell** [comment]
@mstensmo Did you want to resolve the conflicts so we can merge this?  Or is this no longer needed?

**Comment by mstensmo** [comment]
@ledell  This PR is still blocked by https://0xdata.atlassian.net/browse/PUBDEV-3900 . Once it is done we can refresh this one and merge. See Jira https://0xdata.atlassian.net/browse/PUBDEV-4062

**Comment by abal5** [comment]
@h2o-ops please test!

**Pull request merged**